### ISSUE TITLE
[Bug][a11y] Consider increasing the text contrast ratio on the "Ask to allow" option from site permission

### ISSUE THREAD

**Issue by lobontiumira** [mozilla-mobile/fenix#16395]
## Steps to reproduce 
1. Have the Accessibility scanner app installed and opened.
2. In Firefox, go to Settings - Privacy and security - Site permissions and tap on any of the following options: Camera, Location, Microphone, Notification, or Persistent Storage.
3. Scan the opened page.

### Expected behavior
No suggestions are made.

### Actual behavior
The "Ask to allow" option is recommended to be changed:
Text contrast
org.mozilla.fenix:id/ask_to_allow_radio
The item's text contrast ratio is 3.37. This ratio is based on an estimated foreground color of #888888 and an estimated background color of #F9F9FB. Consider using colors that result in a contrast ratio greater than 4.50 for small text, or 3.00 for large text.

### Device information
* Android device: Google Pixel (Android 10)
* Fenix version: 11/3 Nightly

![permission](https://user-images.githubusercontent.com/40694158/98242178-b77caa00-1f74-11eb-9dfa-3cd635382168.png)

**Comment by ivan-suarez** [comment]
Hi, I would like to work on this. However I encountered some problems.

First I reproduced the bug as mentioned above, and had the same results. I then proceeded to calculate the optimum text color using this [contrast calculator](https://contrast-ratio.com/), which I found to be #737373. But when I checked the colors.xml file, the secondary text color value was already #737373, even though the color on the screen was #888888.

Please let me know if I'm doing something wrong.

**Comment by geneukum** [comment]
Hi @ivan-suarez,

Unless I'm missing something, it looks like the color for this 'secondary text' is set here:
https://github.com/mozilla-mobile/fenix/blob/9fee34960098817f1dbdb3c9bea99b0ad011d283/app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsManagePhoneFeatureFragment.kt#L248
rather than in the colors.xml file?

**Comment by priyanka0906** [comment]
@eliserichards may I work on this? Please assign this to me if someone is not working on this.

**Comment by geneukum** [comment]
Just as a heads up, I opened PR https://github.com/mozilla-mobile/fenix/pull/17997 for this issue back in February. I apologize if I've stepped on anyone's toes. Please feel free to decline that PR and to assign the issue to @priyanka0906 if that's desirable here.

**Comment by eliserichards** [comment]
Eek! Sorry it's been sitting around un-reviewed for so long @geneukum. 

This definitely needs UX approval if we are going to change text colors. Tagging @topotropic for thoughts :)

**Comment by topotropic** [comment]
Thanks for the ping, @eliserichards!

@geneukum Thanks for your contribution! could you update the color value for the second line to  
`<color name="photonDarkGrey05">#5B5B66</color>`
Thanks!

**Comment by geneukum** [comment]
Hi @topotropic,

Thanks for the feedback! I am happy to change the color as desired.

In light mode, when the background is #F9F9FB, displaying #5B5B66 text in the foreground should give us a contrast ratio around 6.37. In dark mode, when the background is #1C1B22, displaying #5B5B66 text in the foreground should give us a contrast ratio around 2.54. In this bug, I believe we're trying to get our contrast ratio above 4.50. As such, do you think that it would make sense to display different colors here when in light vs dark mode?

**Comment by topotropic** [comment]
Thanks, @geneukum! 

Please use this color in dark mode for the caption ("Recommended" in the example above)
`    <color name="photonLightGrey50">#BFBFC9</color>`

and for the label ("Ask to allow" in the example above)
`    <color name="photonLightGrey05">#FBFBFE</color>`

**Comment by geneukum** [comment]
Okay, I've updated the commit over in the PR (#17997) to utilize the colors that @topotropic was nice enough to call out. Thanks again!

**Comment by lobontiumira** [comment]
Verified again on the 11/2 Nightly build with Google Pixel (Android 10), and the Accessibility Scanner suggested the following:

![Screenshot_20211102-170103](https://user-images.githubusercontent.com/40694158/139873358-333f71b6-0998-4df5-88a3-67fb7ef05d3d.png)

**Comment by Alexandru2909** [comment]
Accessibility Scanner has no suggestion for any of the Camera, Location, Microphone, Notification, or Persistent Storage screens. Tested on Nightly using Samsung Note 10(Android 12) and Redmi Note 4(Android 7). However, when I analyze the screen using a color picker I get the same color codes, thus the same contrast ratio. Just to be sure the issue is still there, can @mozilla-mobile/qa please verify?

**Comment by geneukum** [comment]
Hey all,

When rebasing my changes, I noticed that the fx_mobile_text_color_secondary color combo (#5B5B66 / #CFCFD8) has been recently added as of commit #4fe3d6db02. This color combination seems very similar to what was suggested by @topotropic from UX above (#5B5B66 / #BFBFC9), and should still give us a good color contrast ratio here. 

Since it has been a little while since we originally looked at this, I wanted to double check -- should we use (#5B5B66 / #BFBFC9) or (#5B5B66 / #CFCFD8)? 

Thanks in advance,
Geordan
